### PR TITLE
Add json based persistors

### PR DESCRIPTION
**Pull request by kjanosz**
Json based persistor with tests that check their correctness.

### CODE REVIEW

**Comment by kjanosz** [comment]
@axel22: As for test that measure performance of new persistor, I would make a new PR with it and include also results of testing new persistors on same large benchmark.
Sounds that good to you?

File: src/main/scala/org/scalameter/persistence/IOStreamPersistor.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+/** Base for persistors that actually write and read [[org.scalameter.History]].
+ *
+ * @tparam I input stream to which [[org.scalameter.History]] is serialized
+ * @tparam O output stream from which [[org.scalameter.History]] is deserialized
+ */
+trait IOStreamPersistor[I <: InputStream, O <: OutputStream] extends Persistor {
+  def path: File
+
+  def fileExt: String
```
Comment (axel22) [inline src/main/scala/org/scalameter/persistence/IOStreamPersistor.scala]: Please either add a more descriptive name (`fileExtension`), or add a ScalaDoc comment.

File: src/main/scala/org/scalameter/persistence/IOStreamPersistor.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  protected def inputStream(file: File): I
+
+  protected def outputStream(file: File): O
+
+  protected def loadFrom(is: I): History
+
+  protected def saveTo(history: History, os: O): Unit
+
+  private def sep: String = File.separator
```
Comment (axel22) [inline src/main/scala/org/scalameter/persistence/IOStreamPersistor.scala]: Same here.

File: src/main/scala/org/scalameter/persistence/SerializationPersistor.scala [hunk 3]
```diff
@@ -3,62 +3,30 @@ package persistence
...
+    override def resolveClass(desc: ObjectStreamClass) = Class.forName(desc.getName)
   }
 
-  def load(context: Context): History = {
-    val scope = context.scope
-    val curve = context.curve
-    val resultdir = context(resultDir)
-    loadHistory(resultdir, scope, curve)
-  }
+  protected def outputStream(file: File): ObjectOutputStream= new ObjectOutputStream(new FileOutputStream(file))
```
Comment (axel22) [inline src/main/scala/org/scalameter/persistence/SerializationPersistor.scala]: Please add a space before the `=`.

**Comment by axel22** [comment]
That sounds good to me - you can put the performance tests in a separate PR.

**Comment by kjanosz** [comment]
Fixed.

**Comment by axel22** [comment]
LGTM

**Pull request merged**